src/shared/openssl-util.c

```diff
@@ -109,6 +110,91 @@ int rsa_pkey_to_suitable_key_size(
         return 0;
 }
 
+static int pkey_generate_ec_key(int nid, EVP_PKEY **ret_ppkey) {
+
+        assert(ret_ppkey);
+        assert_se(nid != NID_undef);
+
+        int r = 0;
+        _cleanup_(EVP_PKEY_CTX_freep) EVP_PKEY_CTX *pctx = NULL;
+        _cleanup_(EVP_PKEY_CTX_freep) EVP_PKEY_CTX *kctx = NULL;
+        _cleanup_(EVP_PKEY_freep) EVP_PKEY *params = NULL;
+
+        pctx = EVP_PKEY_CTX_new_id(EVP_PKEY_EC, NULL);
+        if (pctx == NULL)
```

Comment:
usually, when comparing a pointer to NULL we just write `if (ptcx)`, rely on C's pointer-to-bool downgrade concept.